include/envoy/http/header_map.h

```diff
@@ -465,6 +466,14 @@ class HeaderMap {
    */
   virtual void remove(const LowerCaseString& key) PURE;
 
+  /**
+   * Remove all instances of headers with key matching the supplied regex.
+   * @param key_matcher supplies the regex to match header keys against. Note
+   * that this will be matching against LowerCaseString, so should be using
+   * lowercase characters.
+   */
+  virtual void remove(const std::regex& key_matcher) PURE;
```

Comment:
Should we consider doing a prefix version? I suspect the regex version is going to be slower, even if heavily tuned and basically a prefix?

Reply:
Fine by me.  I'd be inclined to then swap regex for prefix as I have 3 confirmed users of prefixes and 0 that need generic regex :-)

Reply:
+1

Reply:
got inline fixed - one more push and I'll do this swap as well